compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/KnownTargets.cpp

```diff
@@ -693,6 +693,30 @@ StringRef normalizeHIPTarget(StringRef target) {
   return normalizeAMDGPUTarget(target);
 }
 
+std::optional<StringRef> getAMDSKU(TargetAttr target) {
+  StringRef arch = target.getArch();
+
+  if (arch != "gfx942") {
+    return std::nullopt;
+  }
+
+  TargetChipAttr chip = target.getChip();
+  if (chip) {
+    uint32_t wgpCount = chip.getWgpCount();
+    if (wgpCount == 304) {
+      return "mi300x";
+    }
```

Comment:
Thanks for all the fixes around this code. I spend some time thinking about this approach of raising the target attribute back to the sku and wasn't sure if it's a good idea or not. It seems simple and thought that it may be good enough for amdgpu for now, but I found a counterexample: mi325. It has the same number of CUs as mi300, but the performance characteristics are different.

I think that to make this robust, we have to record the sku in the target attribute itself, similar to how we keep the target arch around today.